src/core/public/chrome/nav_group/nav_group_service.ts

```diff
@@ -254,28 +259,34 @@ export class ChromeNavGroupService {
       application.currentAppId$,
       this.getSortedNavGroupsMap$(),
     ]).subscribe(([appId, navGroupMap]) => {
-      if (appId === 'home') {
-        setCurrentNavGroup(undefined);
-        return;
-      }
       if (appId && navGroupMap) {
         const appIdNavGroupMap = new Map<string, Set<string>>();
-        // iterate navGroupMap
-        Object.keys(navGroupMap)
+        let visibleNavGroups: NavGroupItemInMap[] = [];
+        const visibleUseCases = getVisibleUseCases(navGroupMap);
+        if (visibleUseCases.length === 1 && visibleUseCases[0].id === ALL_USE_CASE_ID) {
+          // If the only visible use case is all use case
+          // All the other nav groups will be visible because all use case can visit all of the nav groups.
+          visibleNavGroups = Object.values(navGroupMap);
+        } else {
           // Nav group of Hidden status should be filtered out when counting navGroups the currentApp belongs to
-          .filter((navGroupId) => navGroupMap[navGroupId].status !== NavGroupStatus.Hidden)
-          .forEach((navGroupId) => {
-            navGroupMap[navGroupId].navLinks.forEach((navLink) => {
-              const navLinkId = navLink.id;
-              const navGroupSet = appIdNavGroupMap.get(navLinkId) || new Set();
-              navGroupSet.add(navGroupId);
-              appIdNavGroupMap.set(navLinkId, navGroupSet);
-            });
+          visibleNavGroups = Object.values(navGroupMap).filter(
+            (navGroup) => navGroup.status !== NavGroupStatus.Hidden
+          );
+        }
+        visibleNavGroups.forEach((navGroup) => {
+          navGroup.navLinks.forEach((navLink) => {
+            const navLinkId = navLink.id;
+            const navGroupSet = appIdNavGroupMap.get(navLinkId) || new Set();
+            navGroupSet.add(navGroup.id);
+            appIdNavGroupMap.set(navLinkId, navGroupSet);
           });
```

Comment:
if we pulled `navGroup.navLinks.forEach(...)` into a local function and called it instead of setting `visibleNavGroups`, we could make the two `filter` + `forEach` loops just one.